rust/lance/src/dataset/scanner.rs

```diff
@@ -927,10 +942,11 @@ impl Scanner {
                 // in a deterministic order.
                 false,
             );
-            if let Some(refine_expr) = filter_plan.refine_expr.as_ref() {
+
+            if let Some(expr) = filter_plan.full_expr.as_ref() {
                 // If there is a prefilter we need to manually apply it to the new data
                 let planner = Planner::new(scan_node.schema());
-                let physical_refine_expr = planner.create_physical_expr(refine_expr)?;
+                let physical_refine_expr = planner.create_physical_expr(expr)?;
                 scan_node = Arc::new(FilterExec::try_new(physical_refine_expr, scan_node)?);
             }
```

Comment:
This was the fix for the filter not being applied. We were totally skipping the non-refine part of the `FilterPlan` for the fragments not covered by the vector index.